pkg-arch.patch

```diff
@@ -0,0 +1,39 @@
+--- PKGBUILD    2021-07-07 13:04:52.099980762 +0430
++++ PKGBUILD_customized 2021-07-07 13:09:30.256647408 +0430
+@@ -9,30 +9,25 @@
+ arch=('any')
+ url="https://github.com/ansible/ansible-lint"
+ license=('MIT')
++makedepends=('git' 'python-setuptools' 'python-setuptools-scm' 'python-toml')
+ depends=('python' 'ansible-base' 'python-ruamel-yaml' 'python-pyaml' 'python-rich' 'python-packaging'
+           'python-wcmatch' 'python-enrich' 'python-tenacity')
+ checkdepends=('python-pytest')
+ optdepends=('yamllint: check for yaml syntax mistakes'
+             'ansible: check official ansible collections')
+-source=(https://github.com/ansible/ansible-lint/archive/v${pkgver}/${pkgname}-${pkgver}.tar.gz
+-        disable_use_scm_version.patch)
+-b2sums=('b3a79edd1a77be80cee690b02ef1679af3bf332bb8ba18f98b5a016f0ce4b35289c5b19a9a4fa247bd3210e170a661a6605b3efa011b0934df902c5e4650679a'
+-        'a090862e657bfc29c89b23b2ecc8c585e15f098310b997e39600b6bdf0cd433e1045e2da348ce27dc2d3c0f9ef01950cf0ec5e7b071bc43d85161faabd8a1f60')
+
+-prepare() {
+-  cd ${pkgname}-${pkgver}
+-  # stop setuptools from using the scm version which doesn't exist in this context
+-  patch -Np1 < "${srcdir}"/disable_use_scm_version.patch
+-  sed -i "/^\[metadata\]/a version = ${pkgver}" setup.cfg
++pkgver() {
++  cd ..
++  python setup.py --version
+ }
+
+ build() {
+-  cd ${pkgname}-${pkgver}
++  cd ..
+   python setup.py build
+ }
+
+ package() {
+-  cd ${pkgname}-${pkgver}
++  cd ..
+   PYTHONHASHSEED=0 python setup.py install --root="${pkgdir}" --optimize=1
```

Comment:
Invoking setuptools like this is unrecommended. You should use PEP517 instead. But I wouldn't recommend building sdist from Git. It's better to have it built in a previous job that everything else would depend on. Then, this one should download that artifact and work with it. This way, you'll get to test the thing that actually gets uploaded to PyPI (Git repo may have extra files that won't be included, for example).

Reply:
I'm getting confused about what we're trying to test here, this is the official way following best practices of Arch Packaging Guidelines that it's actually getting used in archlinux, I'm not sure what's the purpose of this test if the way package gets built here differs from the build that actually happens for the package in archlinux.

Reply:
Yep, sdist is the best source for OS-level packaging ecosystems. Fedora, for example, uses sdists as a source and has a set of macros to build it using PEP517. I know most of the distros do something similar.

As for `setup.py`, consider it being there by accident, it may as well be missing. You must use a PEP517 compliant tool which would determine what hooks to run to build a project (it could be something other than setuptools — flit, poetry, a totally custom in-tree build backend etc.)
`setup.py` only exists in this project to accommodate editable installs in old pip versions. I'm hearing that it's no longer necessary in the latest versions so at some point we may consider getting rid of it (because it's becoming just a junk file + external packagers abuse it and hardcode the build backend when they shouldn't).